client/js/controllers/post_main_controller.js

```diff
@@ -153,6 +154,25 @@ class PostMainController extends BasePostController {
         post.save()
             .then(() => {
                 this._view.sidebarControl.showSuccess('Post saved.');
+                // now save the new tags with categories
+                let tagPromises = [];
+                for (let newTag of e.detail.newTags) {
+                    // load the tag that was created during updating the post
+                    tagPromises.push(
+                        Tag.get(newTag.names[0]).then(tag => {
+                                tag.category = newTag.category;
+                                return tag.save();
+                            }
+                        ).then(() => {
+                            e.detail.newTags.removeByName(newTag.names[0]);
```

Comment:
I'm a bit concerned if it's ok to save`e` in closure.
And also if it's ok to create promises to remove from array, while I'm iterating the array.